hub/core/partial_reader.py

```diff
@@ -0,0 +1,29 @@
+from typing import Dict
+from hub.core.storage import LRUCache
+
+
+class PartialReader:
+    def __init__(self, cache: LRUCache, path: str, header_offset: int):
+        self.cache = cache
+        self.path = path
+        self.data_fetched: Dict[tuple[int, int], memoryview] = {}
+        self.header_offset = header_offset
+
+    def __getitem__(self, slice_: slice) -> memoryview:
+        start = slice_.start + self.header_offset
+        stop = slice_.stop + self.header_offset
+        step = slice_.step
+        assert start is not None and stop is not None
+        assert step is None or step == 1
+        slice_tuple = (start, stop)
+        if slice_tuple not in self.data_fetched:
```

Comment:
We should account for partial overlap too right?

Reply:
we're only fetching full samples so there's no chance of overlap